src/masternode-payments.cpp

```diff
@@ -552,35 +568,61 @@ bool CMasternodeBlockPayees::IsTransactionValid(const CTransaction& txNew)
 
     // if we don't have at least 6 signatures on a payee, approve whichever is the longest chain
     if (nMaxSignatures < MNPAYMENTS_SIGNATURES_REQUIRED) return true;
-
+    bool masternodePaymentFound = false;
     BOOST_FOREACH (CMasternodePayee& payee, vecPayments) {
-        bool found = false;
         BOOST_FOREACH (CTxOut out, txNew.vout) {
             if (payee.scriptPubKey == out.scriptPubKey) {
                 if(out.nValue >= requiredMasternodePayment)
-                    found = true;
+                    masternodePaymentFound = true;
                 else
                     LogPrint("masternode","Masternode payment is out of drift range. Paid=%s Min=%s\n", FormatMoney(out.nValue).c_str(), FormatMoney(requiredMasternodePayment).c_str());
             }
         }
 
         if (payee.nVotes >= MNPAYMENTS_SIGNATURES_REQUIRED) {
-            if (found) return true;
-
+            if (!masternodePaymentFound)
+            {
+                CTxDestination address1;
+                ExtractDestination(payee.scriptPubKey, address1);
+                CBitcoinAddress address2(address1);
+
+                if (strPayeesPossible == "") {
+                    strPayeesPossible += address2.ToString();
+                } else {
+                    strPayeesPossible += "," + address2.ToString();
+                }
+                LogPrint("masternode","CMasternodePayments::IsTransactionValid - Missing required payment of %s to %s\n", FormatMoney(requiredMasternodePayment).c_str(), strPayeesPossible.c_str()); 
+            }
+        }
+    }
+    bool charityPaymentRequired = false;
+    bool charityPaymentFound = false;
```

Comment:
charityPaymeentOk = true;

then reset it to false in the "if" and use as charityPaymeentFound

in return masternodePaymeentFound && charityPaymeentOk